packages/dumb/src/components/tooltips/bolus-tooltip/bolus-tooltip.tsx

```diff
@@ -0,0 +1,208 @@
+
+interface BolusTooltipProps {
+  bolus: Bolus | Wizard
+  side: Side
+  position: Position
+  timePrefs: TimePrefs
+}
+
+export const BolusTooltip: FunctionComponent<BolusTooltipProps> = (props) => {
+  const { bolus, position, side, timePrefs } = props
+  const { t } = useTranslation('main')
+
+  const isWizard = bolus.type === DatumType.Wizard
+
+  const bolusData = getBolusFromInsulinEvent(bolus)
+
+  // Common properties
+  const prescriptor = bolusData?.prescriptor
+  const shouldDisplayPrescriptor = prescriptor && prescriptor !== Prescriptor.Manual
+  const bolusSubType = bolusData?.subType
+  const iob = bolusData?.insulinOnBoard
+  const delivered = getDelivered(bolusData as Bolus)
+  const isInterrupted = isInterruptedBolus(bolusData as Bolus)
+  const programmed = getProgrammed(bolusData as Bolus) ?? 0
+  const undeliveredValue = programmed - delivered
+
+  // Wizard-specific properties
+  const carbs = (bolus as Wizard).carbInput
+  const fatMeal = (bolus as Wizard).inputMeal?.fat
+  const isFatMeal = fatMeal === WizardInputMealFat.Yes
+  const inputTime = (bolus as Wizard).inputTime
+  const recommended = getRecommended(bolus as Wizard)
+  const suggested = Number.isFinite(recommended) ? recommended : null
+  const override = programmed - recommended
+  const overrideValue = programmed > recommended ? `+${override}` : `-${override}`
+  const shouldDisplayOverride = Number.isFinite(programmed) && Number.isFinite(recommended) && programmed !== recommended
+  const shouldDisplayRecommended = (isInterrupted || shouldDisplayOverride) && suggested !== null
+
+  const getTitleByBolusType = (bolusType: BolusType): string => {
+    switch (bolusType) {
+      case BolusType.Manual:
+        return t('Manual Bolus')
+      case BolusType.Meal:
+        return t('Meal Bolus')
+      case BolusType.Micro:
+        return t('Micro Bolus')
+    }
+  }
+
+  const getColorByBolusType = (bolusType: BolusType): string => {
+    switch (bolusType) {
+      case BolusType.Manual:
+        return colors.bolusManual
+      case BolusType.Meal:
+        return colors.bolusMeal
+      case BolusType.Micro:
+        return colors.bolusMicro
+    }
+  }
+
+  const bolusType = getBolusType(bolus)
+  const bolusTypeTitle = getTitleByBolusType(bolusType)
+  const color = getColorByBolusType(bolusType)
+
+  return (
+    <Tooltip
+      dateTitle={getDateTitle(bolus, timePrefs)}
+      title={bolusTypeTitle}
+      borderColor={color}
+      position={position}
+      side={side || COMMON_TOOLTIP_SIDE}
+      tailHeight={COMMON_TOOLTIP_TAIL_HEIGHT}
+      tailWidth={COMMON_TOOLTIP_TAIL_WIDTH}
+      tail={DEFAULT_TOOLTIP_TAIL}
+      offset={DEFAULT_TOOLTIP_OFFSET}
+      borderWidth={DEFAULT_TOOLTIP_BORDER_WIDTH}
+      content={
+        <div className={styles.container} id="bolus-tooltip-content">
+          {isWizard && carbs &&
+            <div className={commonStyles.row} id="bolus-tooltip-line-carbs">
```

Comment:
opt:
You could have a components taking `label`, `value` and `units` as optionnal props and display them as this div does it. Less code duplication, but a bit more logic. Do as you prefer.